packages/server/internal/services/favorite.go

```diff
@@ -0,0 +1,112 @@
+package services
+
+import (
+	"fmt"
+
+	"github.com/alexandros-georgantas/platform-go-challenge/internal/helpers"
+	"github.com/alexandros-georgantas/platform-go-challenge/internal/models"
+	"github.com/alexandros-georgantas/platform-go-challenge/internal/serializers"
+	"gorm.io/gorm"
+)
+
+type FavoriteService interface {
+	GetFavorite(uId uint, aId uint) (*models.AssetResponse, error)
+	GetFavorites(uId uint) (serializers.FavoritesResponse, error)
+	AddToFavorites(uId uint, aId uint) (*models.Favorite, error)
+	RemoveFromFavorites(fId uint) (*uint, error)
+}
+
+type favoriteService struct {
+	db gorm.DB
+}
+
+func NewFavoriteService(db gorm.DB) (FavoriteService, error) {
+	return &favoriteService{db: db}, nil
+}
+
+func (fs *favoriteService) GetFavorite(uId uint, aId uint) (*models.AssetResponse, error) {
+	var favorite models.Favorite
+
+	if err := fs.db.Joins("Asset").Where("user_id = ?", uId).First(&favorite, aId).Error; err != nil {
+		return nil, err
+	}
+
+	result, aErr := helpers.AggregateAssetType(favorite.Asset, fs.db)
+
+	if aErr != nil {
+		return nil, aErr
+	}
+
+	return result, nil
+}
+
+func (fs *favoriteService) GetFavorites(uId uint) (serializers.FavoritesResponse, error) {
+	var favorites []models.Favorite
+	fr := models.FavoritesResponse{}
+	pr := serializers.FavoritesResponse{}
+	var count int64
+	pr.Favorites = nil
+	pr.TotalCount = 0
+
+	// TODO: Here first check if we have items in cache and return them, if not proceed with hitting the DB
+	// (take into consideration the pagination params)
+
+	fs.db.Model(&models.Favorite{}).Where("user_id = ?", uId).Count(&count)
+
+	if err := fs.db.Joins("Asset").Where("user_id = ?", uId).Order("asset_id asc").Order("updated_at desc").Find(&favorites).Error; err != nil {
+		return pr, err
+	}
```

Comment:
A better approach could be to group all the logic which interacts with the data layer inside its corresponding entity’s repository, e.g. UserRepository,AssetRepository, etc. by doing so we can abstract the details of interacting with the data layer enabling better modularity.  
The repository pattern could also hide technical choices like which specific ORM is currently in use by just exposing functions like `FindAllAssets()`, `FindAllFavorites()`, etc. This could enable better flexibility in the sense that at any time ORM can change with minimum implications.